service/src/main/java/uk/gov/hmcts/reform/fpl/service/ReplyToMessageJudgeService.java

```diff
@@ -147,13 +147,17 @@ private List<Element<JudicialMessage>> replyToJudicialMessage(UUID selectedJudic
 
                     JudicialMessage judicialMessage = judicialMessageElement.getValue();
 
-                    String sender = judicialMessageReply.getReplyFrom();
+                    String sender = resolveSenderEmailAddress(judicialMessageReply.getSenderType(),
+                        judicialMessageReply.getReplyFrom());
 
                     JudicialMessage updatedMessage = judicialMessage.toBuilder()
                         .dateSent(formatLocalDateTimeBaseUsingFormat(time.now(), DATE_TIME_AT))
                         .updatedTime(time.now())
+                        .senderType(resolveSenderRoleType(judicialMessageReply.getSenderType()))
```

Comment:
trying to understand the helper methods :) guess judicialMessageReply.getSenderType() doesn't return JUDICIARY and that's why we need the helper?

Reply:
Yes. the senderRoleType drop down list doesn’t include Judiciary, as we don't want someone else to mis-claim themselves as Judiciary. 

The judicialMessageReply.senderType will remain empty if a judge is sending the message. 
That's why I have to resolve to role type before storing the new message in about-to-submit stage